pkg/deployment/deployment.go

```diff
@@ -299,7 +299,22 @@ func Deploy(
 		return result, err
 	}
 
-	status.Deployed = true
+	result, err = DeployNovaExternalCompute(
+		ctx,
+		helper,
+		obj,
+		sshKeySecret,
+		inventoryConfigMap,
+		status,
+		networkAttachments,
+		openStackAnsibleEERunnerImage)
+	if err != nil || result.RequeueAfter > 0 {
+		return result, err
+	}
```

Comment:
as soon as CreateOrPatch() called inside, this should immediately exit the reconsiliation loop, without further data manipulations

Reply:
I think you misunderstood the original problem or stale data reconciliation. As far as I understand it is about updating our own instance.Status. When that happens then we have to return immediately. Here in the other hand a different resource is created (or updated)